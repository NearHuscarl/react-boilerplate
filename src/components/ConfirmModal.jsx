import React from 'react';
import Modal from 'react-modal';

const ConfirmModal = (props) => (
	<Modal
		isOpen={props.isOpen}
		onRequestClose={props.onRequestClose}
		contentLabel="Confirm Delete"
		closeTimeoutMS={200}
		className="modal"
		ariaHideApp={false}>
		<h3 className="modal__title">{props.prompt}</h3>
		<button className="button modal__button" onClick={props.onConfirm}>
			yes
		</button>
		<button className="button modal__button" onClick={props.onCancel}>
			no
		</button>
	</Modal>
);

export default ConfirmModal;
