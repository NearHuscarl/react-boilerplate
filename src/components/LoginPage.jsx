import React from 'react';
import { connect } from 'react-redux';
import {
	startLoginWithGoogle,
	startLoginWithFacebook,
	startLoginWithTwitter,
	startLoginWithGithub,
} from '../actions/auth';

export const LoginPage = (props) => (
	<div className="box-layout">
		<div className="box-layout__box">
			<h1 className="box-layout__title">Boilerplate</h1>
			<p>Tag line for app.</p>
			<button
				className="box-layout__button button--red"
				onClick={props.startLoginWithGoogle}>
				<i className="fa fa-google fa-fw" />
				Login with Google
			</button>
			<button
				className="box-layout__button"
				onClick={props.startLoginWithFacebook}>
				<i className="fa fa-facebook fa-fw" />
				Login with Facebook
			</button>
			<button
				className="box-layout__button button--light-blue"
				onClick={props.startLoginWithTwitter}>
				<i className="fa fa-twitter fa-fw" />
				Login with Twitter
			</button>
			<button
				className="box-layout__button button--dark"
				onClick={props.startLoginWithGithub}>
				<i className="fa fa-github fa-fw" />
				Login with Github
			</button>
		</div>
	</div>
);

const mapDispatchToProps = (dispatch) => ({
	startLoginWithGoogle: () => dispatch(startLoginWithGoogle()),
	startLoginWithFacebook: () => dispatch(startLoginWithFacebook()),
	startLoginWithTwitter: () => dispatch(startLoginWithTwitter()),
	startLoginWithGithub: () => dispatch(startLoginWithGithub()),
});

export default connect(
	undefined,
	mapDispatchToProps,
)(LoginPage);
