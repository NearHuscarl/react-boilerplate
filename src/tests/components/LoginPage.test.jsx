import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should render LoginPage correctly', () => {
	const wrapper = shallow(<LoginPage />);
	expect(wrapper).toMatchSnapshot();
});

test('should call startLoginWithGoogle on button click', () => {
	const startLoginWithGoogle = jest.fn();
	const wrapper = shallow(
		<LoginPage startLoginWithGoogle={startLoginWithGoogle} />,
	);
	wrapper
		.findWhere((n) => n.text() === 'Login with Google' && n.type() === 'button')
		.simulate('click');
	expect(startLoginWithGoogle).toHaveBeenCalled();
});

test('should call startLoginWithFacebook on button click', () => {
	const startLoginWithFacebook = jest.fn();
	const wrapper = shallow(
		<LoginPage startLoginWithFacebook={startLoginWithFacebook} />,
	);
	wrapper
		.findWhere(
			(n) => n.text() === 'Login with Facebook' && n.type() === 'button',
		)
		.simulate('click');
	expect(startLoginWithFacebook).toHaveBeenCalled();
});

test('should call startLoginWithTwitter on button click', () => {
	const startLoginWithTwitter = jest.fn();
	const wrapper = shallow(
		<LoginPage startLoginWithTwitter={startLoginWithTwitter} />,
	);
	wrapper
		.findWhere(
			(n) => n.text() === 'Login with Twitter' && n.type() === 'button',
		)
		.simulate('click');
	expect(startLoginWithTwitter).toHaveBeenCalled();
});

test('should call startLoginWithGithub on button click', () => {
	const startLoginWithGithub = jest.fn();
	const wrapper = shallow(
		<LoginPage startLoginWithGithub={startLoginWithGithub} />,
	);
	wrapper
		.findWhere((n) => n.text() === 'Login with Github' && n.type() === 'button')
		.simulate('click');
	expect(startLoginWithGithub).toHaveBeenCalled();
});
