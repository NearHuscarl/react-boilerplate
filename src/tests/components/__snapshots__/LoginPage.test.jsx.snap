// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      It's time to get your expenses under control.
    </p>
    <button
      className="box-layout__button button--red"
    >
      <i
        className="fa fa-google fa-fw"
      />
      Login with Google
    </button>
    <button
      className="box-layout__button"
    >
      <i
        className="fa fa-facebook fa-fw"
      />
      Login with Facebook
    </button>
    <button
      className="box-layout__button button--light-blue"
    >
      <i
        className="fa fa-twitter fa-fw"
      />
      Login with Twitter
    </button>
    <button
      className="box-layout__button button--dark"
    >
      <i
        className="fa fa-github fa-fw"
      />
      Login with Github
    </button>
  </div>
</div>
`;
